gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -242,28 +242,45 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     end
   end
 
-  def self._handle_missing_method_signature(klass, original_method, callee)
+  def self._handle_missing_method_signature(receiver, original_method, callee)
     method_sig = T::Private::Methods.signature_for_method(original_method)
+    if !method_sig
```

Comment:
Doing this instead of raising in all non-alias cases is the fix for **1** from the PR description (refactored slightly from @Morriar's PR for clarity but not meaningfully changed - if we returned a nil method signature we were going to blow up anyway.)